src/main/java/com/commercetools/sync/products/actions/ProductUpdateActionsBuilder.java

```diff
@@ -0,0 +1,228 @@
+package com.commercetools.sync.products.actions;
+
+import com.commercetools.sync.products.ProductSyncOptions;
+import com.commercetools.sync.products.helpers.ProductSyncUtils;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.products.Product;
+import io.sphere.sdk.products.ProductData;
+import io.sphere.sdk.products.ProductDraft;
+import io.sphere.sdk.products.commands.updateactions.AddToCategory;
+import io.sphere.sdk.products.commands.updateactions.ChangeName;
+import io.sphere.sdk.products.commands.updateactions.ChangeSlug;
+import io.sphere.sdk.products.commands.updateactions.SetCategoryOrderHint;
+import io.sphere.sdk.products.commands.updateactions.SetMetaDescription;
+import io.sphere.sdk.products.commands.updateactions.SetMetaKeywords;
+import io.sphere.sdk.products.commands.updateactions.SetMetaTitle;
+import io.sphere.sdk.products.commands.updateactions.SetSearchKeywords;
+import io.sphere.sdk.products.commands.updateactions.SetSku;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonList;
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+import static java.util.stream.Collectors.toList;
+
+public class ProductUpdateActionsBuilder {
+
+    private static final ProductUpdateActionsBuilder productUpdateActionsBuilder;
+
+    // eager singleton
+    static {
+        productUpdateActionsBuilder = new ProductUpdateActionsBuilder();
```

Comment:
Right, I removed this constraint. I was also thinking it'd be good to extract interface for all action builders as providing them as utility static classes is not good. They should only provide one public method for building all actions for specific module.